python_modules/dagster/dagster/_core/definitions/asset_key.py

```diff
@@ -19,7 +21,11 @@ def parse_asset_key_string(s: str) -> Sequence[str]:
 
 
 @whitelist_for_serdes
-class AssetKey(NamedTuple("_AssetKey", [("path", PublicAttr[Sequence[str]])])):
+@record_custom(
+    checked=False,
+    field_to_new_mapping={"parts": "path"},
```

Comment:
i oped to move to `@record` as well to be able to use `field_to_new_mapping` which tersely handles the serialization compat